cognite/preprocessing.py

```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf-8 -*-
+'''Preprocessing module.
+
+This module provides a number of preprocessing methods to further clean the data retrieved from the Cognite API.
+'''
+
+import math
+import warnings
+from functools import reduce
+
+import numpy as np
+import pandas as pd
+
+
+def _merge_dataframes(df1, df2):
+    if df1 is None:
+        return df2
+
+    if df2 is None:
+        return df1
+
+    return pd.merge(df1, df2, on='timestamp', how='outer') \
+        .sort_values(by='timestamp') \
+        .reset_index(drop=True)
+
+
+def merge_list_of_dataframes(dataframes):
+    """Merges together a list of dataframes and creates a time index with evenly spaced intervals.
+
+    Adds NaN to timestamps with missing values.
+
+    Args:
+        dataframes (list(pandas.DataFrame)):  Input dataframes.
+
+    Returns:
+        pandas.DataFrame: Dataframes merged into one with evenly spaced intervals.
+    """
+    df_merged = reduce(lambda x, y: _merge_dataframes(x, y), dataframes)
```

Comment:
Pandas Concat is an alternative here I think. Might be faster.
https://pandas.pydata.org/pandas-docs/stable/generated/pandas.concat.html